data/blog/umdctf-2023/pokeptcha.md

```diff
@@ -0,0 +1,414 @@
+---
+title: UMDCTF 2023 – POKéPTCHA
+date: '2023-05-01'
+draft: false
+authors: ['thebadgod']
+tags: ['UMDCTF 2023', 'Web', 'Reverse', 'VM', 'JavaScript']
+summary: 'Ever wanted to reverse a VM in obfuscated JavaScript?'
+canonical: 'https://spclr.ch/umdctf-pokeptcha'
+---
+
+## POKéPTCHA (Web, 500 Points)
+
+> Team Rocket keeps taking down my website! I’m testing out this new type of captcha, but it doesn’t seem to be working as expected. None of the choices are valid! Can you solve it for me?
+>
+> Author: umasi, sd
+>
+> NOTE: Flag format is `UMDCTF{}`, with the correct answer in the brackets.
+>
+> https://pokeptcha.chall.lol/
+
+### Solve
+
+Opening the website will give us a website where we have to identify a Pokémon based on its silhouette. It’s the classic scene and we are given the choices from the episode, however all of them just trigger the text `Team Rocket detected!`, so let’s see what the correct answer is.
+
+Pressing _F12_ immediatly triggers a debugger statement, which indicates that there is some anti-debugging in place. So for now I disabled the breakpoints to at least see the sources (One could also download the JavaScript source code directly by looking at the networks tab / the HTML).
+
+The JavaScript is obfuscated, however what immediatly looked out is the [`https://github.com/blueimp/JavaScript-MD5/blob/master/js/md5.min.js`](https://github.com/blueimp/JavaScript-MD5/blob/master/js/md5.min.js) comment, which allowed me to identify a big part (everything after the comment) as part of the MD5 algorithm. The big base64 data is printable text, all numbers and `|` characters. The very first function defined looks like this after some deobfuscation:
```

Comment:
> ``[`https://github.com/blueimp/JavaScript-MD5/blob/master/js/md5.min.js`](https://github.com/blueimp/JavaScript-MD5/blob/master/js/md5.min.js)``

Preferably replace the bare URL with something more representative.

Reply:
This is there because source code has that exact line of comment - that's why we want the string to be visible. Otherwise we would normally use some text to replace